src/components/SelectionList/Search/ActionCell.tsx

```diff
@@ -7,28 +7,57 @@ import useLocalize from '@hooks/useLocalize';
 import useStyleUtils from '@hooks/useStyleUtils';
 import useTheme from '@hooks/useTheme';
 import useThemeStyles from '@hooks/useThemeStyles';
+import Navigation from '@navigation/Navigation';
 import variables from '@styles/variables';
+import * as SearchActions from '@userActions/Search';
 import CONST from '@src/CONST';
+import type {TranslationPaths} from '@src/languages/types';
+import ROUTES from '@src/ROUTES';
+import type {SearchTransactionAction} from '@src/types/onyx/SearchResults';
+
+const actionTranslationsMap: Record<SearchTransactionAction, TranslationPaths> = {
+    view: 'common.view',
+    review: 'common.review',
+    done: 'common.done',
+    paid: 'iou.settledExpensify',
+    hold: 'iou.hold',
+    unhold: 'iou.unhold',
+};
 
 type ActionCellProps = {
-    onButtonPress: () => void;
-    action?: string;
+    action?: SearchTransactionAction;
+    transactionID?: string;
+    searchHash: number;
     isLargeScreenWidth?: boolean;
     isSelected?: boolean;
+    goToItem: () => void;
 };
 
-function ActionCell({onButtonPress, action = CONST.SEARCH.ACTION_TYPES.VIEW, isLargeScreenWidth = true, isSelected = false}: ActionCellProps) {
+function ActionCell({action = CONST.SEARCH.ACTION_TYPES.VIEW, transactionID, searchHash, isLargeScreenWidth = true, isSelected = false, goToItem}: ActionCellProps) {
     const {translate} = useLocalize();
-    const styles = useThemeStyles();
     const theme = useTheme();
+    const styles = useThemeStyles();
     const StyleUtils = useStyleUtils();
 
+    const onButtonPress = useCallback(() => {
+        if (!transactionID) {
+            return;
+        }
+
+        if (action === CONST.SEARCH.ACTION_TYPES.HOLD) {
+            Navigation.navigate(ROUTES.TRANSACTION_HOLD_REASON_RHP.getRoute(CONST.SEARCH.TAB.ALL, transactionID, searchHash));
+        } else if (action === CONST.SEARCH.ACTION_TYPES.UNHOLD) {
+            SearchActions.unholdMoneyRequestOnSearch(searchHash, [transactionID]);
```

Comment:
Maybe instead of passing the hash as an argument here, we could just compute it from the route within `unholdMoneyRequestOnSearch` itself. Would that remove the need for passing the hash around?